azure-devops/asan-pipeline.yml

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Build STL targeting x86 and x64, and run extra ASan testing
+
+variables:
+  tmpDir: 'D:\Temp'
+  buildOutputLocation: 'D:\build'
+  # We do not want to build the benchmarks in this pipeline
+  benchmarkBuildOutputLocation: ''
+  benchmarkBuildOutputLocationVar: ''
+  # Restrict to "stlasan" test
+  testSelection: '-R stlasan'
+
+pool:
+  name: 'StlBuild-2023-09-14T1251-Pool'
+  demands: EnableSpotVM -equals true
+
+pr:
+  drafts: false

```

Comment:
Yes. 

I thought it would be nice to make it possible to trigger the pipeline with a PR comment. I believe doing so means we have to have it "triggered" but tell the UI defer running the pipeline until a maintainer posts the magic comment. The value of the comment trigger doesn't justify the extra cost of maintaining that UI configuration, so I'll drop the idea for now.